index.js

```diff
@@ -21,30 +22,20 @@ const exec = (cmd, args) => {
 	).filter(Boolean);
 };
 
-const install = (bin, useYarn) => () => execa(bin, ['install'])
-	.catch(err => {
-		if (err.code === 'ENOENT' && useYarn) {
-			return execa('npm', ['install']);
-		}
-		throw err;
-	});
-
 module.exports = (input, opts) => {
 	input = input || 'patch';
 
 	opts = Object.assign({
 		cleanup: true,
-		publish: true
-	}, opts, {
-		yarn: opts && opts.yarn !== false
-	});
+		publish: true,
+		yarn: hasYarn()
+	}, opts);
```

Comment:
If you would force a lockfile it would fail downstream because it tries to push the tags but not everything is committed yet... So I vote against the possibility of enforcing a lockfile because it would fail...